addons/website_sale/static/src/js/website_sale.editor.js

```diff
@@ -186,14 +365,82 @@ options.registry.WebsiteSaleProductsItem = options.Class.extend({
+        this.$target[0].dataset.ribbonId = widgetValue;
+        this.trigger_up('set_product_ribbon', {
+            templateId: this.productTemplateID,
+            ribbonId: widgetValue || false,
         });
+        const ribbon = this.ribbons[widgetValue] || {name: '', color: '', text_color: '', html_class: ''};
+        const $ribbons = $(`[data-ribbon-id="${widgetValue}"] .o_ribbon:not(.o_wsale_ribbon_dummy)`);
+        $ribbons.text(ribbon.name);
+        $ribbons.attr('class', 'o_ribbon');
+
+        if (!ribbon.html_class) {
+            $ribbons.css('background-color', ribbon.color);
+        } else {
+            $ribbons.addClass(ribbon.html_class);
+            $ribbons.css('background-color', '');
+        }
+        $ribbons.css('color', ribbon.text_color);
+
+        if (!this.ribbons[widgetValue]) {
+            $(`[data-ribbon-id="${widgetValue}"]`).each((index, product) => delete product.dataset.ribbonId);
+        }
+        this.$ribbon.remove();
+        this.$ribbon = this.$('.o_ribbon').clone().addClass('d-none o_wsale_ribbon_dummy').appendTo(this.$target);
+    },
+    /**
+     * @see this.selectClass for params
+     */
+    editRibbon(previewMode, widgetValue, params) {
+        this.saveMethod = 'modify';
+        this._toggleEditingUI(true);
+    },
+    /**
+     * @see this.selectClass for params
+     */
+    createRibbon(previewMode, widgetValue, params) {
+        this.saveMethod = 'create';
+        this.$ribbon.text('Ribbon text');
+        this.$ribbon.addClass('bg-primary');
+        this._toggleEditingUI(true);
+    },
+    /**
+     * @see this.selectClass for params
+     */
+    async deleteRibbon(previewMode, widgetValue, params) {
+        const {ribbonId} = this.$target[0].dataset;
+        this.trigger_up('delete_ribbon', {id: ribbonId});
+        await this._rerenderXML();
+        await this.setRibbon(false, '');
+    },
+    /**
+     * @see this.selectClass for params
+     */
+    async saveRibbon(previewMode, widgetValue, params) {
+        const ribbon = {
+            'name': this.$ribbon.text(),
+            'color': this.$ribbon[0].style.backgroundColor,
+            'text_color': this.$ribbon[0].style.color,
+            'html_class': this.$ribbon.attr('class').replace(/(d-none)|(o_wsale_ribbon_dummy)|(o_ribbon)/g, '').trim(),
+        };
+        ribbon.id = this.saveMethod === 'modify' ? parseInt(this.$target[0].dataset.ribbonId) : Date.now();
+        this.trigger_up('set_ribbon', {ribbon: ribbon});
+        await this._rerenderXML();
+        await this.setRibbon(false, ribbon.id);
+    },
+    /**
+     * @see this.selectClass for params
+     */
+    setRibbonColor(previewMode, widgetValue, params) {
+        this.selectStyle.call(Object.assign({}, this, {$target: this.$ribbon}), previewMode, widgetValue, params);
+    },
+    /**
+     * @see this.selectClass for params
+     */
+    setRibbonName(previewMode, widgetValue, params) {
+        this.$ribbon.text(widgetValue);
```

Comment:
Could be a `data-select-property="" data-property-name="textContent"`, less code

Reply:
As discussed, since the commit that implements that isn't merged yet, and doesn't implement a computeWidgetState method which I need, I'll leave it as is for now.